### ISSUE TITLE
SVG's initial coordinate system

### ISSUE THREAD

**Issue by master-toma** [cduck/drawSvg#11]
Hello,
Is it possible to setup SVG viewport to SVG's initial coordinate system? Where 0;0 is the top-left corner. Now I can see, that by default 0;0 is the bottom-left corner. 
Thank you

**Comment by cduck** [comment]
Not really.  drawSvg was designed for the coordinates x and y to increase rightward and upward.

However, you can still place the origin at the top left.  The y-coordinates will just be negative:
```
import drawSvg as draw
d = draw.Drawing(4,4, origin=(0,-4))

d.append(draw.Circle(0,-0,1, fill='red'))
d.append(draw.Circle(2,-2,0.75, fill='green'))
d.append(draw.Circle(3,-3,0.5, fill='green'))
d.append(draw.Text('Text at (2,1)',0.5, 2,-1, text_anchor='middle'))

d.setRenderSize(h=200)
d
```
![output1](https://user-images.githubusercontent.com/2476062/69498630-cbeda100-0ee1-11ea-8531-247135472e58.png)

Another option if you don't want to negate the y-coordinate is to use an SVG scale transform to flip the y-axis (this mirrors the text unless a second transform is applied to it):
```
import drawSvg as draw
d = draw.Drawing(4,4, origin=(0,0))

g = draw.Group(transform='scale(1,-1) translate(0,4)')
g.append(draw.Circle(0,0,1, fill='red'))
g.append(draw.Circle(2,2,0.75, fill='green'))
g.append(draw.Circle(3,3,0.5, fill='green'))
g.append(draw.Text('Text at (2,1)',0.5, 0,0, text_anchor='middle',
                   transform='scale(1,-1) translate(2,1)'))
d.append(g)

d.setRenderSize(h=200)
d
```
![output2](https://user-images.githubusercontent.com/2476062/69498631-d0b25500-0ee1-11ea-8129-f9a6d4ff52c6.png)

**Comment by master-toma** [comment]
Thank you a lot, first method worked fine with me, just wondered if there some more convenient for such a use case. And thank you for your library

**Comment by JeroenBos** [comment]
> d = draw.Drawing(4,4, origin=(0,0))
> g = draw.Group(transform='scale(1,-1) translate(0,4)')

That combination still doesn't do it.

To have the origin be upper-left corner and such that the viewbox ends up being `0 0 width height`, use
```
d = draw.Drawing(width, height, origin=(0,-height))
g = draw.Group(transform='scale(1,-1)')
```

**Comment by cduck** [comment]
@JeroenBos If you run my second example does it not draw a red circle in the top left?  Can you post the full code you ran that didn't work as expected, and the output image?

**Comment by cgevans** [comment]
Would you consider a pull request to allow the user to choose, at a module level, whether to use an upward-y or downward-y coordinate system?  It doesn't seem like it would be that difficult to allow this, and I have a draft of the change working.

**Comment by cduck** [comment]
Yes I would but the design is important.  I have considered this but to allow interoperability of files that use each convention, I would strongly prefer it use some kind of file-based configuration, not global.

For example, if you implement methods `drawSvg.upwardY()` and `drawSvg.downwardY()` this case where a dependency uses one convention but use use another would break:

some_package.py
```python
import drawSvg as draw
drawSvg.upwardY()
def draw_something():
    ...
```
my_code.py
```python
import drawSvg as draw
import some_package
drawSvg.downwardY()
def main():
    some_package.draw_something()
def draw_something_else():
    ...
```
-----
Here is a solution I've considered that avoids the issue of conflicting global configuration:
some_package.py
```python
import drawSvg.upwardY as draw
def draw_something():
    ...
```
my_code.py
```python
import drawSvg.downwardY as draw
import some_package
def main():
    some_package.draw_something()
def draw_something_else():
    ...
```
Two different submodules, `upwardY` and `downwardY` with subclasses of each of the drawing elements configured with upward or downward y.

What is your opinion on this?

**Comment by cduck** [comment]
Hi @master-toma, @JeroenBos, @cgevans, @mgritter, @popolinneto, @iongion, joachimheintz, @aufarah.  You all have commented here about drawSvg's reversed-Y coordinate system so I'd appreciate if you could take a look at drawsvg 2.0 (#89) that restores the original SVG coordinate system (and optionally allows it to be reversed).

Thanks for all your feedback about how the reversed-Y made the library a bit confusing and inconsistent.  Help me catch anything I missed that would require additional API changes before I publish version 2.0 (comment on #89).  I hope to publish version 2 of the package by Feb 25th.

**Comment by cduck** [comment]
Fixed in [version 2.0](https://pypi.org/project/drawsvg/2.0.0/).
